lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -50,6 +52,107 @@ using TaggedFK = TaggedFunctionKind;
 void TaggedFK::getInvalidations(const Context &Ctx,
                                 TargetsList &ToRemove,
                                 const GlobalTupleTreeDiff &Diff) const {
+  return;
+}
+
+using IsolatedFK = IsolatedFunctionKind;
+void IsolatedFK::getInvalidations(const Context &Ctx,
+                                  TargetsList &ToRemove,
+                                  const GlobalTupleTreeDiff &Diff) const {
+  using PM = PathMatcher;
+  using Binary = model::Binary;
+
+  const auto &CurrentModel = getModelFromContext(Ctx);
+  if (not Ctx.containsReadOnlyContainer(BinaryCrossRelationsRole))
+    return;
+
+  static constexpr auto BCRR = BinaryCrossRelationsRole;
+  const auto &Container = Ctx.getReadOnlyContainer<FileContainer>(BCRR);
+
+  auto MaybeContainerPath = Container.path();
+  revng_assert(MaybeContainerPath.has_value());
+  auto MaybeBuffer = llvm::MemoryBuffer::getFile(*MaybeContainerPath);
+  revng_assert(MaybeBuffer);
+  yaml::Input YAMLInput((*MaybeBuffer)->getBuffer());
+
+  yield::CrossRelations Relations;
+  YAMLInput >> Relations;
+
+  const auto *ModelDiff = Diff.getAs<Binary>();
+  if (not ModelDiff)
+    return;
+
+  const Binary &Model = *CurrentModel.get();
+  auto EntryMatcher = PM::create<Binary>("/Functions/*/Entry").value();
+  auto FSOMatcher = PM::create<Binary>("/Types/RawFunctionType-*/"
+                                       "FinalStackOffset")
+                      .value();
+  auto PresRegsMatcher = PM::create<Binary>("/Types/RawFunctionType-*/"
+                                            "PreservedRegisters")
+                           .value();
```

Comment:
I'm afraid these won't do. We're not considering `CABIFunctionType`.

I'm starting to think that the best approach, at this point, is to ignore the diff and just do:

```c++
for (const model::Function &NewFunction : NewModel.Functions) {
  const auto &OldFunction = OldModel.Functions.at(NewFunction.Entry);
  Layout Old = makeLayout(OldFunction);
  Layout New = makeLayout(NewFunction);

  if (Old.FSO != New.FSO) {
    // ...
  }

}
```

In the future we could refine things reasoning on the diff to be more efficient.

Reply:
I don't understand. `CABIFunctionType` does not come with `FinalStackOffset`, the user cannot even change it, hence, why would we ever want to pass to `Layout` and check `FinalStackOffset` for `CABIFunctionType` prototypes?